### ISSUE TITLE
Setting array to [] still cause calls from template

### ISSUE THREAD

**Issue by ErikWallin** [ractivejs/ractive#2069]
Hi,

Below is a complete example of a thing that I don't understand in ractive. I have an array of items which is first rendered correctly. However, when I empty that array (ractive.set('items', []);), unnecessary calls are then being made (func: {{ func(this) }}). My expectations was that those calls should not happen since items is an empty array and the template should not trigger calls.

To run the example, just copy the code into an html file and run in a browser. Watch the console where func logs 'undefined' three times when emptying the items array.

Can someone explain why that happens? I think it is a defect.

I use ractive 0.7.3, but have tried older versions as well.

```
<!doctype html>
<html lang='en-GB'>
<head>
  <meta charset='utf-8'>
  <title>Ractive test</title>
</head>

<body>
  <h1>Ractive test</h1>

  <div id='container'></div>

  <script id='template' type='text/ractive'>
    <p>Hello, {{name}}!</p>

    <ul>
      {{#items}}
        <li>
          {{#a}}
            func: {{ func(this) }}
            f: {{ this.f() }}
            d: {{ this.d }}
          {{/a}}
          b: {{ b }}
        </li>
      {{/items}}
    </ul>
  </script>

  <script src='http://cdn.ractivejs.org/latest/ractive.min.js'></script>

  <script>
    var ractive = new Ractive({
      el: '#container',

      template: '#template',

      data: function () {
        return {
          name: 'world',

          items: [
            {
              a: {
                f: function() { console.log('f1'); return 'f1';},
                d: '22'
              },
              b: '2'
            },
            {
              a: {
                f: function() { console.log('f2'); return 'f2';},
                d: '44'
              },
              b: '4'
            },
            {
              a: {
                f: function() { console.log('f3'); return 'f3';},
                d: '66'
              },
              b: '6'
            }
          ],

          func: function(obj) { console.log(obj); return obj;}
        };
      }
    });

    setTimeout(function() {
      ractive.set('items', []);
    }, 2000);
    
  </script>
</body>
</html>
```

**Comment by Madgvox** [comment]
I am fairly certain that this is the same effect as explained in #2023. For an explanation as to what's happening below the covers, see [this comment](https://github.com/ractivejs/ractive/issues/2023#issuecomment-118792948). This behavior is indeed not intended, and will be fixed in 0.8 with #1740.

**Comment by Rich-Harris** [comment]
Actually this isn't quite fixed in 0.8 yet! :grimacing: I've put it in a fiddle for reference: http://jsfiddle.net/rich_harris/vr3veefo/

**Comment by Rich-Harris** [comment]
This is now fixed
